src/Symfony/Component/Messenger/Bridge/Doctrine/Tests/Transport/PostgreSqlConnectionTest.php

```diff
@@ -42,6 +47,68 @@ public function testUnserialize()
         $connection->__wakeup();
     }
 
+    public function testListenOnConnection()
+    {
+        $driverConnection = $this->createMock(\Doctrine\DBAL\Connection::class);
+
+        $driverConnection
+            ->expects(self::any())
+            ->method('getDatabasePlatform')
+            ->willReturn(new PostgreSQLPlatform());
+
+        $driverConnection
+            ->expects(self::any())
+            ->method('createQueryBuilder')
+            ->willReturn(new QueryBuilder($driverConnection));
+
+        $wrappedConnection = new class() {
+            private $notifyCalls = 0;
+
+            public function pgsqlGetNotify()
+            {
+                ++$this->notifyCalls;
+
+                return false;
+            }
+
+            public function countNotifyCalls()
+            {
+                return $this->notifyCalls;
+            }
+        };
+
+        // dbal 2.x
+        if (interface_exists(Result::class)) {
```

Comment:
I had to do this weird thing to test against dbal 2.x and dbal 3.x. I'm not proud of it but i do not have a better idea

Reply:
You could maybe duplicate the test to remove the if (and skip if not using the good version). But IMHO that's good enough as-is.